src/App.tsx

```diff
@@ -1,56 +1,92 @@
-import React from 'react';
+import React, { useMemo, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person>();
+  const [inputFocus, setInputFocus] = useState(false);
+
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const personInfo = `${selectedPerson?.name} (${selectedPerson?.born} - ${selectedPerson?.died})`;
+
+  const handleQueryChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+    delay: number,
+  ) => {
+    setQuery(event.target.value);
+
+    setTimeout(() => {
+      setAppliedQuery(event.target.value);
+    }, delay);
+  };
+
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer.filter(person => {
+      const name = person.name.toLowerCase();
+      const lowerQuery = appliedQuery.toLowerCase().trim();
+
+      return name.includes(lowerQuery);
+    });
+  }, [appliedQuery]);
+
+  const onSelected = (
```

Comment:
```suggestion
  const handlePersonSelected = (
```
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364